app/api/controllers/validators/resolvers/ResolveNino.scala

```diff
@@ -34,4 +34,12 @@ object ResolveNino extends Resolver[String, Nino] {
       Invalid(List(withError(error, NinoFormatError, path)))
   }
 
+  def isValid(nino: String): Boolean = nino != null && hasValidPrefix(nino) && ninoRegex.matches(nino)
```

Comment:
Can we update `ResolveNino` in full to ensure that `isValid` method is fully consistent with the `apply` method (i.e. `apply` uses `isValid`). Otherwise we run the risk of having anomalous edge cases.

You should also update `ResolveNinoSpec`.